dwave/embedding/polynomialembeddor.py

```diff
@@ -0,0 +1,1222 @@
+        return score
+
+    def maxCliqueWithRectangle(self, R, maxCWR):
+        """This does the dirty work for :func:`nativeCliqueEmbed`.  Not meant to be
+        understood or called on its own.  Guaranteed to maintain the inductive
+        hypothesis that ``maxCWR`` is optimal.  We put in the tiniest amount of
+        effort to return a uniform random choice of a maximum-sized native
+        clique embedding.
+
+        INPUTS:
+            R (tuple): the rectangle specified as a tuple ``(xmin,xmax,ymin,ymax)``
+
+            maxCWR (dict): the dictionary we're building inductively which maps
+            ``R -> (score, ell, r, num)`` where score is the size of the best
+            clique with working rectangle ``R``, and ``ell`` and ``r`` indicate how to
+            reconstruct that clique: add the maximum line bundle in ``ell``, and
+            recursively examine the working rectangle ``r``.  The parameter ``num`` is the number
+            of cliques with ``R`` as a working rectangle and size equal to ``score``.
+
+        OUTPUT:
+            score (int): the score for the returned clique (just ``len(clique)``
+            in the class :class:`eden_processor`; may differ in subclasses)
+
+            best (tuple): a tuple ``(score, ell, parent, num)`` to be stored in
+            ``maxCWR[R]``.
+
+            * score: as above
+            * ell: ``(x0,x1,y0,y1)`` defines the unit cells in an ell-shaped
+                region: ``(x0,y1),...,(x0,y0),...,(x1,y0)``
+            * parent: the rectangle ``R1`` for which the clique generated
+                recursively by looking up ``maxCWR[R1]`` as described above.
+            * num: the number of partial native clique embeddings with ``R``
+                as a working rectangle and a score of ``score``.
+
+        """
+
+        (xmin, xmax, ymin, ymax) = R
+        best = nothing = 0, None, None, 1
+        bestscore = None
+        count = 0
+        N = self.N
+
+        Xlist = (xmin, xmax, xmin + 1, xmax), (xmax, xmin, xmin, xmax - 1)
+        Ylist = (ymin - 1, ymax, ymin - 1,
+                 ymax), (ymax + 1, ymin, ymin, ymax + 1)
+        XY = [(X, Y) for X in Xlist for Y in Ylist if 0 <= Y[2] <= Y[3] < N]
+
+        bests = []
+        for X, Y in XY:
+            x0, x1, nxmin, nxmax = X
+            y0, y1, nymin, nymax = Y
+            r = nxmin, nxmax, nymin, nymax
+            try:
+                rscore, rell, rparent, nr = maxCWR[r]
+            except:
+                rscore, nr = None, 1
+            score = self._combine_clique_scores(
+                rscore, (y0, xmin, xmax), (x0, nymin, nymax))
+            if bestscore is None or score > bestscore:
+                bestscore = score
+                count = 0
+            if score == bestscore:
+                count, best = _accumulate_random(
+                    count, nr, best, (score, (x0, x1, y0, y1), r, nr))
+
+        return bestscore, best
+
+    def nativeCliqueEmbed(self, width):
```

Comment:
Methods should all be in snake case or all camel case. It's weird to see both in a single file.